ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoSuite.scala

```diff
@@ -443,6 +445,28 @@ private[dao] trait JdbcLedgerDaoSuite extends AkkaBeforeAndAfterAll with JdbcLed
       implicit ec: ExecutionContext): Future[(Offset, LedgerEntry.Transaction)] =
     store(divulgedContracts = Map.empty, offsetAndTx)
 
+  protected final def storeSync(commands: Vector[(Offset, LedgerEntry.Transaction)])(
+      implicit ec: ExecutionContext): Future[Vector[(Offset, LedgerEntry.Transaction)]] = {
+
+    import scalaz.Free
+    import scalaz.syntax.traverse._
+    import scalaz.std.vector._
+    import scalaz.std.scalaFuture._
+
+    val storeDelayed = (a: (Offset, LedgerEntry.Transaction)) => Free.liftFU(store(a))
+
+    // force synchronous future processing with Free monad
+    // to provide the guarantees that all transactions persisted in the specified order
+    val xs: Free[Future, Vector[(Offset, LedgerEntry.Transaction)]] =
+      commands.traverse(storeDelayed)
+
+    xs.foldMap(noopNaturalTransformation[Future])
+  }
+
+  private def noopNaturalTransformation[F[_]] = new scalaz.NaturalTransformation[F, F] {
+    override def apply[A](fa: F[A]): F[A] = fa
+  }
+
```

Comment:
@S11001001: I could not find a `noop` natural transformation in scalaz. Is there one?

Reply:
Yep; suggested above.